### PR TITLE
Rename find() to findall()

### PR DESCRIPTION
**Pull request by nalimilan**
`findall` is more explicit and fits well in the series of `findfirst`/`findlast` and `findnext`/`findprev`. This change got some support at https://github.com/JuliaLang/julia/issues/10593.

### CODE REVIEW

**Comment by JeffBezanson** [comment]
Looks like there's a doc reference issue.

File: NEWS.md [hunk 1]
```diff
@@ -367,11 +367,12 @@ This section lists changes that do not have deprecation warnings.
...
 
-  * `find` now returns the same type of indices as `keys`/`pairs` for `AbstractArray`,
-    `AbstractDict`, `AbstractString`, `Tuple` and `NamedTuple` objects ([#24774]).
+  * `find` has been renamed to `findall`, and now returns the same type of indices
+    as `keys`/`pairs` for `AbstractArray`, `AbstractDict`, `AbstractString`, `Tuple`
+    and `NamedTuple` objects ([#24774], [#25545]).
     In particular, this means that it returns `CartesianIndex` objects for matrices
     and higher-dimensional arrays instead of linear indices as was previously the case.
-    Use `LinearIndices(a)[find(f, a)]` to compute linear indices.
+    Use `LinearIndices(a)[findall(f, a)]` to compute linear indice
```
Comment (KristofferC) [inline NEWS.md]: Accidentally cut off this sentence?

Comment (nalimilan) [reply NEWS.md]: Ah, thanks, that's an editor bug, it's not the first time I get this...

**Comment by nalimilan** [comment]
Merge?

**Pull request merged**